src/main/java/com/FoodyBuddy/Dao/Impl/CountryDAOImpl.java

```diff
@@ -1,55 +1,63 @@
-package com.FoodyBuddy.Dao.Impl;
+package com.FoodyBuddy.DAO.Impl;
 
-import java.util.List;
+import com.FoodyBuddy.DAO.Country;
+import com.FoodyBuddy.Model.Country;
 
+import java.util.List;
 import org.hibernate.Session;
-import org.hibernate.SessionFactory;
-import org.hibernate.Transaction;
-
-import com.FoodyBuddy.Dao.CountryDAO;
-import com.FoodyBuddy.Model.Country;
-import com.FoodyBuddy.Model.Seller;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
 
 
+@Repository
 public class CountryDAOImpl implements CountryDAO {
+	
+	private static final Logger logger = LoggerFactory.getLogger(CountryDAOImpl.class);
 
-	private SessionFactory sessionFactory;
-
-    public void setSessionFactory(SessionFactory sessionFactory) {
-        this.sessionFactory = sessionFactory;
-    }
-    
-	public void save(Country c) {
-		Session session = this.sessionFactory.openSession();
-		Transaction tx = session.beginTransaction();
-		session.persist(c);
-		tx.commit();
-		session.close();
-	}
+	private Session session = null;
 	
+	public CountryDAOImpl(Session session){
+		this.session = session;
+	}
+
+	@Override
+	public void insertCountry(Country country) {
+		session.persist(country);
+		logger.info("Country saved successfully, Country Details = " + country);
+	}
+
+	@Override
+	public void updateCountry(Country country) {
+
+		session.update(country);
+		logger.info("Country updated successfully, Country Details = " + country );
+	}
+
 	@SuppressWarnings("unchecked")
-	public List<Country> list() {
-		Session session = this.sessionFactory.openSession();
-		List<Country> CountryList = session.createQuery("from Country").list();
-		session.close();
-		return CountryList;
+	@Override
```

Comment:
remove all @Override tags as we are implementing an interface and not extending class.